task/src/main/java/com/flipkart/flux/impl/task/TaskExecutor.java

```diff
@@ -78,7 +78,12 @@ protected Event run() throws Exception {
 		}
 		final Object returnObject = result.getKey();
 		if (returnObject != null) {
-			return new Event(outputeEventName, returnObject.getClass().getCanonicalName(), Event.EventStatus.triggered, stateMachineId, objectMapper.writeValueAsString(returnObject), MANAGED_RUNTIME);
+            Event event = (Event) returnObject;
+            event.setName(outputeEventName);
```

Comment:
I thought about creating another class, but it would be a overkill as we can get away the approach I used :). If we send only a string from JVM task, we can't get the type of the class in executor.